lib/java_buildpack/framework/dynatrace_one_agent.rb

```diff
@@ -46,83 +46,84 @@ def compile
+        unless environment.key?(DT_APPLICATION_ID)
+          environment_variables.add_environment_variable(DT_APPLICATION_ID, application_id)
         end
 
-        environment_variables.add_environment_variable(RUXIT_HOST_ID, host_id) unless environment.key?(RUXIT_HOST_ID)
+        environment_variables.add_environment_variable(DT_HOST_ID, host_id) unless environment.key?(DT_HOST_ID)
+        environment_variables.add_environment_variable(DT_TENANT, credentials[ENVIRONMENTID])
+        environment_variables.add_environment_variable(DT_TENANTTOKEN, tenanttoken)
+        environment_variables.add_environment_variable(DT_CONNECTION_POINT, endpoints)
       end
 
       protected
 
       # (see JavaBuildpack::Component::VersionedDependencyComponent#supports?)
       def supports?
-        @application.services.one_service? FILTER, [ENVIRONMENTID, TENANT], [APITOKEN, TENANTTOKEN]
-      end
-
-      def supports_apitoken?
-        credentials = @application.services.find_service(FILTER)['credentials']
-        credentials[APITOKEN] ? true : false
+        !service.nil?
       end
 
       private
 
-      FILTER = /ruxit|dynatrace/
+      FILTER = /dynatrace/
 
-      RUXIT_APPLICATION_ID = 'RUXIT_APPLICATIONID'.freeze
+      DT_APPLICATION_ID = 'DT_APPLICATIONID'.freeze
 
-      RUXIT_HOST_ID = 'RUXIT_HOST_ID'.freeze
+      DT_HOST_ID = 'DT_HOST_ID'.freeze
 
-      SERVER = 'server'.freeze
+      DT_TENANT = 'DT_TENANT'.freeze
 
-      TENANT = 'tenant'.freeze
+      DT_TENANTTOKEN = 'DT_TENANTTOKEN'.freeze
 
-      TENANTTOKEN = 'tenanttoken'.freeze
+      DT_CONNECTION_POINT = 'DT_CONNECTION_POINT'.freeze
 
       APITOKEN = 'apitoken'.freeze
 
       APIURL = 'apiurl'.freeze
 
       ENVIRONMENTID = 'environmentid'.freeze
 
-      ENDPOINT = 'endpoint'.freeze
-
-      private_constant :FILTER, :RUXIT_APPLICATION_ID, :RUXIT_HOST_ID, :SERVER, :TENANT, :TENANTTOKEN, :APITOKEN
-      private_constant :ENVIRONMENTID, :ENDPOINT, :APIURL
+      private_constant :FILTER, :DT_APPLICATION_ID, :DT_HOST_ID
+      private_constant :DT_TENANT, :DT_TENANTTOKEN, :DT_CONNECTION_POINT
+      private_constant :ENVIRONMENTID, :APITOKEN
+
+      def service
+        candidates = @application.services.select do |candidate|
+          (
+            candidate['name'] =~ FILTER ||
+            candidate['label'] =~ FILTER ||
+            candidate['tags'].any? { |tag| tag =~ FILTER }
+          ) &&
+          candidate['credentials'][ENVIRONMENTID] && candidate['credentials'][APITOKEN]
+        end
 
-      def agent_dir
-        @droplet.sandbox + 'agent'
+        candidates.one? ? candidates.first : nil
       end
 
       def agent_path
-        libpath = agent_dir + 'lib64/liboneagentloader.so'
-        libpath = agent_dir + 'lib64/libruxitagentloader.so' unless File.file?(libpath)
-        libpath
+        technologies = JSON.parse(File.read(@droplet.sandbox + 'manifest.json'))['technologies']
```

Comment:
You probably don't want to read this manifest multiple times.  I'll clean up in a polishing commit.